core/sugartypes.ml

```diff
@@ -369,7 +369,10 @@ and regex =
   | Splice    of phrase
   | Replace   of regex * replace_rhs
 and clause = Pattern.with_pos * phrase
-and funlit = Pattern.with_pos list list * phrase
+and funlit = NormalFunlit of normal_funlit | MatchFunlit of match_funlit
+and match_funlit = Pattern.with_pos list list * match_body
+and match_body = (Pattern.with_pos * phrase) list
```

Comment:
I think this should be `(Pattern.with_pos list * phrase) list`, where if the (curried) function takes several lists of patterns the first argument is the list of curried patterns rather than a single tuple pattern. Indeed, the data of a single case should be the same as `funlit`, so you could just define `match_body` as `funlit list`.

Alternatively, forbidding `match` on functions that take multiple parameter lists and specializing to 
```
and funlit = NormalFunlit of normal_funlit | MatchFunlit of match_funlit
and match_funlit = Pattern.with_pos list * match_body
and match_body = (Pattern.with_pos * phrase) list
```
would be fine.

Reply:
I think the second one looks more sensible. What is the example of a curried function with pattern matching?